tests/SBRPTests/SBRPTests.cs

```diff
@@ -0,0 +1,57 @@
+using System;
+using System.Runtime.InteropServices;
+using Xunit;
+
+namespace SBRPTests;
+
+public class ReferencePackagesTest : IClassFixture<SBRPFixture>, IDisposable
+{
+
+    public string WorkingDirectory { get; set; }
+    public ReferencePackagesTest(SBRPFixture fixture)
+    {
+        WorkingDirectory = fixture.WorkingDirectory;
+    }
+
+    [Theory(Timeout = 300000)]    
+    [MemberData(nameof(PackageData.Data), MemberType = typeof(PackageData))]
+    public void VerifySBRP(string package, PackageType type)
+    {
+        switch (type)
+        {
+            case PackageType.Reference:
+                if (RuntimeInformation.IsOSPlatform(OSPlatform.Windows))
```

Comment:
There is duplication between the window/linux as well as reference/text paths.  For instance the invocation of generatre.cm/sh as well as the validation logic.  To increase maintainability of this code, please refactor to eliminate this duplication.